src/charm.py

```diff
@@ -136,16 +140,14 @@ def _on_install(self, _: typing.Any) -> None:
         self.haproxy_service.install()
         self.unit.status = ops.MaintenanceStatus("Waiting for haproxy to be configured.")
 
-    @validate_config_and_tls(defer=False, block_on_tls_not_ready=False)
```

Comment:
At this point, shouldn't we merge `_on_config_changed`, `_on_certificate_available`, `_on_http_backend_available`, `_on_http_backend_removed` and all other `reconcile` wrappers for readability?